### DISCUSSION TITLE
Share Score in Web for bigger Screens

### DISCUSSION THREAD

**Discussion by MrFootwork** [matco/badminton#24] [Ideas]
If the players can track their own scores, you don't need score judges anymore. But then you would need to be able to display the score on bigger screens.

In our local club we use such a score tracker as a web app. But you need someone off-court to track the score. The score is tracked via any browser. Therefor you need at least a phone to track, which is unpractical on court. The current score can be shared among all club members via any browser and also serves as a display at the side of the court. Something like that in combination with this Garmin app would be a dream for me.

[Here](https://app.dibvm.de/vs/) the web app we use.

**Comment by matco** [comment]
Thanks for your contribution. I think it would be possible to push the score to your web app each time it is updated. More technically, we can configure a URL in the settings of the badminton application and push a JSON payload to the URL with the current score. The web app you use may need to be updated to accept this payload.

The main problem is to find a way to identify the match you are currently playing. We can either have a custom URL for each match (and you will need to configure it in the settings of the badminton application before the beginning of a new match) or find a way to set the ID of the match directly on the watch. How do you currently identify the match? Does the web app provide an ID that you enter on your phone?

In addition to that, I think the HTTP request should be authenticated, so you cannot update the score of another match.

Depending on the level of integration, this development could be time-consuming, and may require modifications in the code of your web app. It may also be quite specific (I don't know if the application you are using is used by other clubs).

**Reply by MrFootwork** [reply]
If you get into portrait mode, you can access an URL to the current match. It can be accessed from the burger menu appearing only in portrait mode. The app also provides a button to share the URL in WhatsApp.

You are right with the specificity. This feature would only benefit my club :( Maybe something else will come up in the future to make it more feasible.

**Reply by matco** [reply]
If you don't mind about updating the URL of the match (in the settings of the Badminton application) before the beginning of each match, it's quite feasible on my side. But I won't send a specific payload for the application of your club, so it will need to be updated as well.

Still, I think it would be very difficult for users to understand this feature.

While there is no easy way to input data on the watch (more than selecting A, B or C), it will be difficult to have this feature working reliably for every user. The best path I can see now is to use the microphone (available in the latest watches) to listen for a "sound message" containing the ID of the match, generated by the application of your club. It could also be possible to develop a companion app running on a phone that would build the bridge between the Badminton app on the watch and the application of your club.

**Reply by MrFootwork** [reply]
The developer of our club's web score tracker is starting to make efforts to promote it nationwide. He wants to convince the German Badminton Federation (DBV) to use it for its members. I like the companion app idea. I ask our club developer, if we could create something like that to bridge the gap.

**Reply by MrFootwork** [reply]
> ...The best path I can see now is to use the microphone (available in the latest watches) to listen for a "sound message" containing the ID of the match, generated by the application of your club. ...

This could be made possible, if our score tracker would use more speakable words as IDs. We could use city names instead of those cryptic hashes, e.g. Berlin, Paris, London.
